src/main/java/hudson/remoting/Callable.java

```diff
@@ -40,4 +47,46 @@
      * or throws some exception.
      */
     V call() throws T;
+
+    /**
+     * Gets a channel for the operation inside callable.
+     * @return Channel Instance
+     * @throws ChannelStateException Channel is not associated with the thread
+     * @since TODO
+     */
+    @Nonnull
+    default Channel getChannelOrFail() throws ChannelStateException {
+        final Channel ch = Channel.current();
+        if (ch == null) {
+            // This logic does not prevent from improperly serializing objects within Remoting calls.
+            // If it happens in API calls in external usages, we wish good luck with diagnosing Remoting issues
+            // and leaks in ExportTable.
+            //TODO: maybe there is a way to actually diagnose this case?
+            final Thread t = Thread.currentThread();
+            throw new ChannelStateException("The calling thread " + t + " has no associated channel. "
+                    + "The current object " + this + " is " + SerializableOnlyOverRemoting.class +
+                    ", but it is likely being serialized/deserialized without the channel");
+        }
+        return ch;
+    }
+
+    /**
+     * Gets an open channel, which is ready to accept commands.
+     *
+     * It is a convenience method for cases, when a callable needs to invoke call backs on the master.
+     * In such case the requests will be likely failed by {@linkplain UserRequest} logic anyway, but it is better to fail fast.
+     * 
+     * @return Channel instance
+     * @throws ChannelStateException The channel is closing down or has been closed.
```

Comment:
Could specify `ChannelClosedException` for this case.

Reply:
yup